File: backend/src/main/java/com/example/backend/Model/Dto/FindByRequestDTO.java
package com.example.backend.Model.Dto;

import com.example.backend.Utility.TableRequest;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class FindByRequestDTO {
	private TableRequest tableRequest;
	private ProcessLogsFilterDTO processLogsFilterDTO;
	private CarsFiltersDTO carsFiltersDTO;
	private QualityChecksFiltersDTO qualityChecksFiltersDTO;
	private CarsPartsFiltersDTO carsPartsFiltersDTO;
	private MachineUsedFiltersDTO machineUsedFiltersDTO;
	private PartProductionFiltersDTO partProductionFiltersDTO;
	private MachineFiltersDTO machineFiltersDTO;
	private UserAllFiltersDTO userAllFiltersDTO;
	private MachineAllFiltersDTO machineAllFiltersDTO;
}

File: frontend/src/app/_model/_dto/find-by-request-dto.ts
import { ChangePage } from '../_common/change-page';
import { SortPage } from '../_common/sort-page';
import { CarsFiltersDTO } from './cars-filter-dto';
import { CarsPartsFilterDTO } from './cars-parts-filter-dto';
import { MachineAllFiltersDTO } from './machine-all-filter.dto';
import { MachineFiltersDTO } from './machine-filters-dto';
import { MachineUsedFiltersDTO } from './machine-used-filters-dto';
import { PartProductionFiltersDTO } from './part_production-filter-dto';
import { ProcessLogsFilterDTO } from './process-log-filter-dto';
import { QualityChecksFiltersDTO } from './quality-check-filter-dto';
import { UserAllFiltersDTO } from './user-all-filters-dto';

export interface FindByRequestDTO {
  tableRequest?: TableRequest;
  processLogsFilterDTO?: ProcessLogsFilterDTO;
  carsFiltersDTO?: CarsFiltersDTO;
  qualityChecksFiltersDTO?: QualityChecksFiltersDTO;
  carsPartsFiltersDTO?: CarsPartsFilterDTO;
  machineUsedFiltersDTO?: MachineUsedFiltersDTO;
  partProductionFiltersDTO?: PartProductionFiltersDTO;
  machineFiltersDTO?: MachineFiltersDTO;
  userAllFiltersDTO?: UserAllFiltersDTO;
  machineAllFiltersDTO?: MachineAllFiltersDTO;
}

export interface TableRequest {
  changePage: ChangePage;
  sortPage: SortPage;
}
